subprojects/docs/src/docs/userguide/authoring-builds/introduction/plugins.adoc

```diff
@@ -0,0 +1,579 @@
+----
+apply(plugin = "org.barfuin.gradle.taskinfo")
+apply<MyPlugin>()
+----
+|
+|===
+
+[[sec:plugins_block]]
+== 1. Applying plugins using the `plugins{}` block
+
+The plugin DSL provides a concise and convenient way to declare plugin dependencies.
+
+[[sec:binary_plugins]]
+The plugins block configures an instance of link:{javadocPath}/org/gradle/plugin/use/PluginDependenciesSpec.html[`PluginDependenciesSpec`]:
+
+[source, build.gradle.kts]
+----
+plugins {
+    application                                     // by name
+    java                                            // by name
+    id("java")                                      // by id
+    id("org.jetbrains.kotlin.jvm") version "1.9.0"  // by id
+}
+----
+
+Core Gradle plugins are unique in that they provide short names, such as `java` for the core link:{javadocPath}/org/gradle/api/plugins/JavaPlugin.html[JavaPlugin].
+
+To apply a core plugin, the short *name* can be used:
+
+====
+include::sample[dir="snippets/plugins/dsl/kotlin", files="build.gradle.kts[tags=use-plugin]"]
+include::sample[dir="snippets/plugins/dsl/groovy", files="build.gradle[tags=use-plugin]"]
+====
+
+All other binary plugins must use the fully qualified form of the plugin id (e.g., `com.github.foo.bar`).
+
+To apply a community plugin from http://plugins.gradle.org[Gradle plugin portal], the fully qualified *plugin id*, a globally unique identifier, must be used:
+
+====
+include::sample[dir="snippets/plugins/dsl/kotlin", files="build.gradle.kts[tags=use-community-plugin]"]
+include::sample[dir="snippets/plugins/dsl/groovy", files="build.gradle[tags=use-community-plugin]"]
+====
+
+See link:{javadocPath}/org/gradle/plugin/use/PluginDependenciesSpec.html[`PluginDependenciesSpec`] for more information on using the Plugin DSL.
+
+[[plugins_dsl_limitations]]
+=== Limitations of the plugins DSL
+
+The plugins DSL provides a convenient syntax for users and the ability for Gradle to determine which plugins are used quickly.
+This allows Gradle to:
+
+* Optimize the loading and reuse of plugin classes.
+* Provide editors with detailed information about the potential properties and values in the build script.
+
+However, the DSL requires that plugins be defined statically.
+
+There are some key differences between the `plugins {}` block mechanism and the "traditional" `apply()` method mechanism.
+There are also some constraints and possible limitations.
+
+[[sec:constrained_syntax]]
+==== Constrained Syntax
+
+The `plugins {}` block does not support arbitrary code.
+
+It is constrained to be idempotent (produce the same result every time) and side effect-free (safe for Gradle to execute at any time).
+
+The form is:
+
+====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+plugins {
+    `«plugin id»`                                             // <1>
+    id(«plugin id»)                                           // <2>
+    id(«plugin id») version «plugin version» [apply «false»]  // <3>
```

Comment:
🤔 I think the `<<plugin id>>` and `<<false>>` placeholders here make this hard to understand.